### PR TITLE
Add support for aarch64

### PR DESCRIPTION
**Pull request by andriish**
- Add support for aarch64
- Add CI that runs on Linux aarch64

Note: on aarch64, the float128 is equivalent to native long double and quadmath library does not exist.

### CODE REVIEW

**Comment by andriish** [comment]
Hi @scarrazza, could you, please, have a look?

Nowadays people make a lot of claims how beneficial it is to do calculations on ARM, but totally forget to invest in the support of software on ARM.

Best regards,

Andrii

**Review by scarrazza — APPROVED** [review]
Dear @andriish, thanks for this, it looks good to me.

**Comment by matthewfeickert** [comment]
This is great @andriish. With this we can also get Linux `aarch64` (and I think also macOS arm64) builds working for the conda-forge distribution: https://github.com/conda-forge/qcdloop-feedstock

(c.f. https://github.com/conda-forge/qcdloop-feedstock/pull/8).

**edit**: @andriish Out of curiosity, have you patched `aarch64` support into other libraries that use `libquadmath`? I know that this is also an issue with `ninja` (https://github.com/peraro/ninja) (c.f. https://github.com/hep-packaging-coordination/packaging-hep-simulation-stack/issues/3)

File: CMakeLists.txt [hunk 1]
```diff
@@ -66,7 +66,7 @@ configure_file(
   "${PROJECT_SOURCE_DIR}/src/qcdloop.pc.in"
   "${PROJECT_SOURCE_DIR}/src/qcdloop.pc"
   )
-
+IF(${CMAKE_SYSTEM_PROCESSOR} MATCHES "x86_64")
```
Comment (matthewfeickert) [inline CMakeLists.txt:69]: @andriish Doesn't this assume only the existence of `x86` and `aarch64`? Won't this break `ppc64le` builds now?

**Pull request merged**